R/methods-epi_df.R

```diff
@@ -383,3 +383,36 @@ arrange_canonical.epi_df <- function(x, ...) {
     dplyr::relocate(dplyr::all_of(keys), .before = 1) %>%
     dplyr::arrange(dplyr::across(dplyr::all_of(keys)))
 }
+
+#' Aggregate an `epi_df` object
+#'
+#' Aggregates an `epi_df` object by the specified group columns, summing the
+#' `value` column, and returning an `epi_df`. If aggregating over `geo_value`,
+#' the resulting `epi_df` will have `geo_value` set to `"total"`.
+#'
+#' @param .x an `epi_df`
+#' @param value_col character name of the column to aggregate
+#' @param group_cols character vector of column names to group by
+#' @return an `epi_df` object
+#'
+#' @export
+aggregate_epi_df <- function(.x, value_col = "value", group_cols = "time_value") {
```

Comment:
question: why `aggregate_epi_df` instead of making an `aggregate.epi_df` implementation?

suggest: rename `value_col` to `counts_col` or something like that, or rename this function `sum_epi_df` or something like that

suggest: support multiple count columns and make it `counts_cols` or something like that

Reply:
> why aggregate_epi_df instead of making an aggregate.epi_df implementation
just a placeholder name, didn't think about it too much. i tend to think about dispatch when i have 2 or more types i want to dispatch over and here i just have one. just `aggregate` seems too generic, though i admit `aggregate_epi_df` isn't much better. maybe `group_by_aggregate`?

Reply:
question: how do i generalize this to allow a character vector in sum_cols?
```R
dplyr::summarize(!!(sum_cols) := sum(!!sym(sum_cols)))
```